src/main.rs

```diff
@@ -130,21 +130,14 @@ async fn check_weekday_pdf(day: Weekdays, pdf_getter: Arc<SubstitutionPDFGetter<
 		}
 	};
 
-	for class in classes {
-		if let Some(new_substitutions) = new_schedule.get_substitutions(class.as_str()) {
-			if let Some(old_schedule) = &old_schedule {
-				if let Some(old_substitutions) = old_schedule.get_substitutions(class.as_str()) {
-					if new_substitutions != old_substitutions {
-						discord.notify_users_for_class(class.as_str(), day).await?;
-					}
-				}
-			} else {
-				discord.notify_users_for_class(class.as_str(), day).await?;
-			}
+	if let Some(old_schedule) = old_schedule_option {
+		if !new_schedule.get_entries().keys().all(|k| old_schedule.get_entries().contains_key(k)) {
```

Comment:
No comparison wether there was actually a change in schedule (Not comparing old != new)